### ISSUE TITLE
NetBSD buildbot

### ISSUE THREAD

**Issue by krytarowski** [nodejs/build#903]
I'm asking for a NetBSD 8.0(beta) buildbot to serve as a CI machine at least for libuv.

CC: @cjihrig

**Comment by refack** [comment]
Hello @krytarowski,

Can you give some background on accomplishing this:
1. Do you know a hosting provider that hosts NetBSD images?
2. What is NetBSD's package manager, and how do we get `git` `python2` and a new enough `gcc` (If NetBSD uses glibc, then one that is build on `glibc >= 2.12`)

Optional: Best help you can provide us is writing an ansible playbook, that setups up a machine (like the ones in `/ansible/` - https://github.com/nodejs/build/blob/master/ansible/playbooks/jenkins/worker/create.yml or https://github.com/nodejs/build/blob/master/setup/freebsd/ansible-playbook.yaml

**Comment by gdams** [comment]
I know that [RootBSD](https://www.rootbsd.net/) hosts NetBSD images

**Comment by krytarowski** [comment]
1. amazon, gce should work; I use for personal purposes https://www.serveraptor.com/
2. pkgsrc - the same as SmartOS;

```
PKG_PATH=http://cdn.netbsd.org/pub/pkgsrc/packages/NetBSD/amd64/8.0_current/All/ pkg_add git python27
```

8.0 ships with GCC 5.4, there is a newer version in pkgsrc like gcc7 (7.1), gcc6 (6.4)... The one from base should be preferred.

I'm not literate in ansible unfortunately.

Changes compared with FreeBSD:

```
ansible_python_interpreter: "/usr/bin/env python"
```

->

```
ansible_python_interpreter: "/usr/pkg/bin/python2.7"
```


This replaced the above command with `pkg_add`:

```
pkg install -U -y {{ item }}
```

**Comment by maclover7** [comment]
Are there any providers willing to donate NetBSD machines? Otherwise I think this should be closed until a more formal plan be written up

**Comment by krytarowski** [comment]
What are the requirements?

**Comment by krytarowski** [comment]
I can run it myself, but I'm not ansible-literate.

**Comment by maclover7** [comment]
Closing for now as something that would be nice to have, but seems to not currently be within our means. If someone wants to tackle this, please feel free to reopen.

**Comment by krytarowski** [comment]
So, what's the option to make libuv a supported tier platform? @cjihrig

**Comment by rvagg** [comment]
1. people to maintain the code
2. infrastructure to integrate it into our CI
3. TSC agreement that we can bump it up into the supported list (likely contingent on 1 & 2)

Re infrastructure, we have some limitations:
 * Trusted: so we try to run as much as possible in IaaS providers or in locations that we have some kind of direct connection with where we can establish trust
 * Stable: because any time a platform goes down our entire CI infra suffers, as do collaborators who are trying to test code, they shouldn't have to think about the platforms it's running on until there's a test failure. We've even been trimming platforms to improve stability, so we really care about this.
 * Gratis: we've built our entire infra platform on donated resources and our preference is to continue. This means that AWS is not an option until we can convince them to donate (a possibility but we haven't been able to do this yet. GCP is not something we have used at all but I know that Google is likely willing to donate resources if we asked nicely. DigitalOcean, Rackspace, Joyent, SoftLayer, Azure are all strong partners that we can potentially expand capacity on.

People resource can't be understated, however. It's probably not going to be enough to have 1 person to call on when something goes wrong and nobody else understands. We need people to maintain the infra and to help triage and fix code bugs when they arise. The fewer people we have and the less responsive they are the lower likelihood that we're ever going to get to supported platform status.
